### ISSUE TITLE
Formatting and removing trailing whitespace

### ISSUE THREAD

**Issue by bonimy** [roice3/MagicCube5D#3]
* I am using a Visual Studio extension that removes trailing whitespace in code. I've noticed several lines have it and I would
  like to remove it if that's okay.
* Tabs and spaces are mixed in the source files. I would like to replace all tabs with four space indents.
* I'm not sure if there is a defined coding style for this project. If not, I would like to propose Google's clang-formatted styles,
  with 88 character line limits. Here is my proposed .clang-format file:

```
---
Language: Cpp
BasedOnStyle: Google
ColumnLimit: 88
IndentWidth: 4
AccessModifierOffset: -4
SortIncludes: true
ConstructorInitializerIndentWidth: 8
ContinuationIndentWidth: 8
DerivePointerAlignment: false
MaxEmptyLinesToKeep: 2
...
```

**Comment by roice3** [comment]
I'm definitely ok with format normalization, and I'm not even going to argue much on the details because I don't really work on MC5D anymore. The only item I take issue with (and would reject on my current projects) is replacing tabs with spaces. I know it is personal preferences, but I much prefer tabs over spaces for indention because of the way I use my keyboard. But like I said, I won't argue much.

If you'd like, I can elevate your privileges on this project so that I don't need to be a bottleneck for merges.

**Comment by bonimy** [comment]
Yeah that'd be good. There have been things I've wanted to change on this and M120Cell.

**Comment by bonimy** [comment]
Resolved with 07c8139..2be6f30
